bpo-41394: State special _ in appendix of tutorial

wyz23x2 opened the pull request:
This patch states the special variable ``_`` in interactive mode and also includes an example.


<!-- issue-number: [bpo-41394](https://bugs.python.org/issue41394) -->
https://bugs.python.org/issue41394
<!-- /issue-number -->

wyz23x2 committed:
Update appendix.rst

ericvsmith commented on Doc/tutorial/appendix.rst:
```diff
@@ -31,6 +31,26 @@ Typing an interrupt while a command is executing raises the
+-------------------------
+
+There is a special variable ``_`` in interactive mode. It stores the result of the 
+last evaluation (*except* ``None``), stored in the :mod:`builtins` module.
```
I think "last evaluation" should be "last evaluated expression".

The "except None" part should be clearer, and say that _ isn't changed if the last evaluated expression evaluated to None.

I'm not sure what the part about builtins is trying to say. Can you clarify?

And with this change, we probably need a NEWS entry, so I'm going to remove the skip news label.

wyz23x2 committed:
Fixed hyper to match "tut"

wyz23x2 committed:
Update appendix.rst

Made the statement more clear.
Added example stating _ is not defined at the start.

wyz23x2 committed:
Create 2020-07-27-23-05-02.bpo-41394.BEos8A.rst

wyz23x2 committed:
bpo-41394: Document '_' in interpreter tutorial

wyz23x2 committed:
Update 2020-07-27-23-05-02.bpo-41394.BEos8A.rst

wyz23x2 committed:
bpo-41394: State special _ in appendix of tutorial

ericvsmith commented on Doc/tutorial/appendix.rst:
```diff
@@ -31,6 +31,32 @@ Typing an interrupt while a command is executing raises the
+Special ``_`` variable
+-------------------------
+
+There is a special variable ``_`` in interactive mode, stored in the :mod:`builtins` 
```
Is it really true that _ is stored in builtins? I played around with it a while, and couldn't verify that. And even if it's true, is it important? I don't think so. Do you have another reference to _ being in builtins?

wyz23x2 replied:
This was stated in bpo-41394 too.
https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers

ericvsmith replied:
Interesting. I still think it's unimportant for the tutorial and should be removed.

wyz23x2 replied:
OK. Working on it.

ericvsmith commented on Doc/tutorial/appendix.rst:
```diff
@@ -31,6 +31,32 @@ Typing an interrupt while a command is executing raises the
+
+There is a special variable ``_`` in interactive mode, stored in the :mod:`builtins` 
+module. It stores the result of the last evaluated expression (*except* ``None``
+— it remains unchanged).
```
I think this would read better as: "It stores the result of the last evaluated expression. If the last expression evaluated to ``None``, then the value of ``_`` is not updated."

wyz23x2 replied:
Agree.

ericvsmith reviewed (commented):
See comments.
